packages/apollo-voyager-keycloak/src/schemaDirectives/hasRole.ts

```diff
@@ -0,0 +1,68 @@
+import { SchemaDirectiveVisitor } from 'graphql-tools'
+import { defaultFieldResolver } from 'graphql'
+import { ForbiddenError } from 'apollo-server-express'
+// import newInternalServerError from '???' // need to figure out where this comes from
+import pino from 'pino' // also need to figure out where this comes from
+import Joi from 'joi'
+
+const log = pino()
```

Comment:
Do they recommend to setup instance for each file or we can have global logger?

Reply:
I haven't seen any specific recommendation but because we're looking to provide things like context logging, we should probably provide some way to plug a logger in. Either for our framework to pass it down, or for the user to pass it down.

Reply:
Frameworks tend to use debug package. Pino etc. are usually used in end user app